### DISCUSSION TITLE
False Detection for CVE-2023-25330 in com.baomidou:mybatis-plus

### DISCUSSION THREAD

**Discussion by kubegut** [aquasecurity/trivy#6287] [False Detection]
### IDs

CVE-2023-25330

### Description

Scanning a custom Docker image finds a vulnerability by the maven artifact com.baomidou:mybatis-plus:3.5.5

Library                                  | Vulnerability  | Severity | Status  | Installed Version | Fixed Version | Title
-----------------------------------------|----------------|----------|---------|-------------------|---------------|--------------------------------------------------------------------------------------------------------
com.baomidou:mybatis-plus (my-app.jar) | CVE-2023-25330 | CRITICAL | affected | 3.5.5             |               | MyBatis-Plus vulnerable to SQL injection via TenantPlugin [More info](https://avd.aquasec.com/nvd/cve-2023-25330)

However, [gitlab](https://advisories.gitlab.com/advisory/advmaven_org_mybatis_mybatis_CVE_2023_25330.html) & [github ](https://github.com/advisories/GHSA-32qq-m9fh-f74w) advisory reports show that the affected versions for com.baomidou:mybatis-plus are below 3.5.3.1, yet it (version 3.5.5) is still being reported by Trivy.

Reference issues:
- https://github.com/aquasecurity/trivy/discussions/5623#discussioncomment-8243936
- [mybatis-plus - CVE-2023-25330 (Critical) detected in mybatis-plus-3.5.3.1.jar](https://github.com/baomidou/mybatis-plus) 

Any help would be appreciated. Thanks.

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Container Image

### Scanner

Vulnerability

### Debug Output

```bash
.
```

### Version

```bash
Version: 0.49.1
```

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [X] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by DmitriyLewen** [comment]
Hello @kubegut 
Thanks for your report!

Created https://github.com/aquasecurity/trivy-db/issues/391 for this task.
Regards, Dmitriy
